File: Ejemplos/Comunicacion/Python/FifoB.py
import os

BYTES = 1024

# Intentar abrir el FIFO para lectura
try:
    fifo = os.open("./fifo", os.O_RDONLY)
    
    # Leer datos del FIFO (hasta BYTES bytes)
    data = os.read(fifo, BYTES)
    
    # Imprimir los datos leídos después de decodificar
    print(data.decode())  
    
    # Cerrar el FIFO después de leer
    os.close(fifo)

except OSError as e:
    print(f"Error al abrir o leer el FIFO: {e}")

File: Ejemplos/Comunicacion/C/fifoB.c
#include<stdlib.h>
#include<stdio.h>
#include<fcntl.h>
#include<unistd.h>

struct persona
{
    int dni;
    char nombre[30];	
};

int main (int argc, char *argv[])
{
    int r = open("./fifo", O_RDONLY);

    struct persona alumno;
    if (read(r, &alumno, sizeof(struct persona)) == -1)
    {
        perror("Error en read()");
        return EXIT_FAILURE;
    }

    printf("%d\n",alumno.dni);	
    printf("%s\n",alumno.nombre);	

    close(r);	
    
    return EXIT_SUCCESS;
}
